src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/CNNScoreVariants.java

```diff
@@ -285,6 +279,61 @@ public void apply(final VariantContext variant, final ReadsContext readsContext,
         sendBatchIfReady();
     }
 
+    @Override
+    public void afterFirstPass() {
+        if (waitforBatchCompletion) {
+            pythonExecutor.waitForPreviousBatchCompletion();
+        }
+        if (curBatchSize > 0) {
+            executePythonCommand();
+            pythonExecutor.waitForPreviousBatchCompletion();
+        }
+
+        pythonExecutor.sendSynchronousCommand("tempFile.close()" + NL);
+        pythonExecutor.terminate();
+
+        try {
+            scoreScan = new Scanner(scoreFile);
+            vcfWriter = createVCFWriter(new File(outputFile));
+            scoreScan.useDelimiter("\\n");
+            writeVCFHeader(vcfWriter);
+        } catch (IOException e) {
+            throw new GATKException("Error when trying to write annotated VCF.", e);
+        }
+
+    }
+
+    @Override
+    protected void secondPassApply(VariantContext variant, ReadsContext readsContext, ReferenceContext referenceContext, FeatureContext featureContext) {
+        String sv = scoreScan.nextLine();
+        String[] scoredVariant = sv.split("\\t");
+
+        if (variant.getContig().equals(scoredVariant[CONTIG_INDEX])
+                && Integer.toString(variant.getStart()).equals(scoredVariant[POS_INDEX])
+                && variant.getReference().getBaseString().equals(scoredVariant[REF_INDEX])
+                && variant.getAlternateAlleles().toString().equals(scoredVariant[ALT_INDEX])) {
+
+            final VariantContextBuilder builder = new VariantContextBuilder(variant);
+            if (scoredVariant.length > KEY_INDEX) {
+                builder.attribute(scoreKey, scoredVariant[KEY_INDEX]);
+            }
+            vcfWriter.add(builder.make());
+
+        } else {
+            String errorMsg = "Score file out of sync with original VCF. Score file has:" + sv;
+            errorMsg += "\n But VCF has:" + variant.toStringWithoutGenotypes();
+            throw new GATKException(errorMsg);
+        }
+    }
+
+    @Override
+    public void closeTool() {
+        logger.info("Done scoring variants with CNN.");
+        if ( vcfWriter != null ) {
+            vcfWriter.close();
+        }
```

Comment:
Should test the Scanner for null and also close that if not.

Reply:
done